src/api/OlsApi.ts

```diff
@@ -93,26 +94,63 @@ export class OlsApi {
     return params;
   }
 
-  private buildParamsForSearch(queryParams: SearchQueryParams, paginationParams: PaginationParams, contentParams?: ContentParams, parameter?: string) {
+  private buildParamsForSearch(queryParams: SearchQueryParams, paginationParams: PaginationParams, contentParams?: ContentParams, parameter?: string, useLegacy=true) {
+
     const params: any = {
-      q: queryParams.query,
-      exact: queryParams.exactMatch,
-      obsoletes: queryParams.showObsoleteTerms,
+
     };
 
-    if (queryParams.groupByIri) {
+    if (useLegacy){
+
+      params.exact = queryParams.exactMatch,
+      params.obsoletes = queryParams.showObsoleteTerms,
+      params.q = queryParams.query
+
+      if (queryParams.groupByIri) {
       params.groupField = queryParams.groupByIri;
-    }
+      }
 
-    if (queryParams.types) {
-      params.type = queryParams.types;
-    }
+      if (queryParams.types) {
+        params.type = queryParams.types;
+      }
+
+      if (queryParams.ontology) {
+        params.ontology = queryParams.ontology;
+      }
+    } else {
+      params.search = queryParams.query
+
+      // @todo not in ols4 v2 API?
```

Comment:
I think OLS4 does it differently. By default the field named "isDefiningOntology" is false. In each search result item is stated that the terms come from which ontology (definedBy) and "Also In" which other ontologies (appearsIn). If you set "isDefiningOntology=true" then it returns only the terms from the original ontologies.

https://www.ebi.ac.uk/ols4/api/v2/entities?search=data item extraction from journal article&size=5&lang=en&exactMatch=false&includeObsoleteEntities=false&isDefiningOntology=true

Reply:
Regarding the "groupByIri"

Reply:
ah okay, thanks for the clarification. So the question is, what do we want as search results:
- only terms within defining ontology
- all terms within different ontologies
I don't know what results we got with "groupByIri=true" - would it be analogous to "isDefiningOntology=true"?

Reply:
Yes kind of. when groupByIri is given in ols3, it groups the result with the same Iri to one that helps us to build "Also In" part. in V2, the info exist in each result "appearsIn" and "definedBy" metadata. So we do not need groupByIri (and I think it is not supported anymore)

This is of course my understanding since there is no documentation. 

Regarding the first question, I have no idea. It is open for discussion.